gui/RespiraConnectedDevice.cpp

```diff
@@ -0,0 +1,83 @@
+#include "../common/third_party/nanopb/pb_common.h"
+#include "../common/third_party/nanopb/pb_decode.h"
+#include "../common/third_party/nanopb/pb_encode.h"
+#include "connected_device.h"
+#include <QSerialPort>
+
+class RespiraConnectedDevice : public ConnectedDevice {
+
+  RespiraConnectedDevice(QString portName, uint32_t write_timeout_ms,
+                         uint32_t interframe_timeout_ms)
+      : m_serial(new QSerialPort()), m_writeTimeout_ms(write_timeout_ms),
```

Comment:
When is the QSerialPort deleted?

Reply:
destructor added to close it.

Reply:
There might be a more Qt-idiomatic way to deal with cleanup if ConnectedDevice inherited from QObject, as already does QSerialPort. The parent destroys all children, as described [here](https://doc.qt.io/qt-5/qobject.html#QObject). Likewise, it seems PeriodicClosure might exist as a [QThread](https://doc.qt.io/qt-5/qthread.html), which would give you the signals/slots mechanism (which you may or may not need?).  I also see the merits of using the possibly more familiar std machinery over Qt's. Upon further examination of the code, I suspect this is irrelevant to how we intend this to work. Just wanted to make you aware of these tools in case Qt is less familiar to you guys.